core/src/main/scala/org/apache/spark/storage/TachyonBlockManager.scala

```diff
@@ -95,8 +94,23 @@ private[spark] class TachyonBlockManager() extends ExternalBlockManager with Log
   override def putBytes(blockId: BlockId, bytes: ByteBuffer): Unit = {
     val file = getFile(blockId)
     val os = file.getOutStream(WriteType.TRY_CACHE)
-    os.write(bytes.array())
-    os.close()
+    try {
+      os.write(bytes.array())
+      os.close()
```

Comment:
I understand that. That is not my point. Any Error for example is not caught by this. This is why a close happens on the finally path by convention.